Python/Product/Debugger/Debugger/DebugEngine/AD7Engine.cs

```diff
@@ -659,92 +710,31 @@ public sealed class AD7Engine : IDebugEngine2, IDebugEngineLaunch2, IDebugProgra
-                                }
-                                break;
-                            case BreakSystemExitZero:
-                                if (Boolean.TryParse(setting[1], out value) && value) {
-                                    debugOptions |= PythonDebugOptions.BreakOnSystemExitZero;
-                                }
-                                break;
-                            case DebugStdLib:
-                                if (Boolean.TryParse(setting[1], out value) && value) {
-                                    debugOptions |= PythonDebugOptions.DebugStdLib;
-                                }
-                                break;
-                            case DirMappingSetting:
-                                string[] dirs = setting[1].Split('|');
-                                if (dirs.Length == 2) {
-                                    if (dirMapping == null) {
-                                        dirMapping = new List<string[]>();
-                                    }
-                                    Debug.WriteLine(String.Format("Mapping dir {0} to {1}", dirs[0], dirs[1]));
-                                    dirMapping.Add(dirs);
-                                }
-                                break;
-                            case InterpreterOptions:
-                                interpreterOptions = setting[1];
-                                break;
-                            case AttachRunning:
-                                attachRunning = Convert.ToBoolean(setting[1]);
-                                break;
-                            case WebBrowserUrl:
-                                _webBrowserUrl = HttpUtility.UrlDecode(setting[1]);
-                                break;
-                            case EnableDjangoDebugging:
-                                if (Boolean.TryParse(setting[1], out value) && value) {
-                                    debugOptions |= PythonDebugOptions.DjangoDebugging;
-                                }
-                                break;
-                        }
-                    }
-                }
+                ParseOptions(options);
             }
 
-            _engineCreated = _programCreated = false;
-            _loadComplete.Reset();
+            Send(new AD7CustomEvent(VsPackageMessage.SetDebugOptions, this), AD7CustomEvent.IID, null, null);
+
+            // If this is a windowed application, there's no console to wait on, so disable those flags if they were set.
+            if (_debugOptions.HasFlag(PythonDebugOptions.IsWindowsApplication)) {
+                _debugOptions &= ~(PythonDebugOptions.WaitOnNormalExit | PythonDebugOptions.WaitOnAbnormalExit);
+            }
 
             Guid processId;
-            if (attachRunning && Guid.TryParse(exe, out processId)) {
+            if (_debugOptions.HasFlag(PythonDebugOptions.AttachRunning) && Guid.TryParse(exe, out processId)) {

```

Comment:
What if the flag says attach and the guid is invalid? Probably shouldn't fall back to launching in that case.